File: Python/Collection/MultiThreading.py
import threading
import time

# Funktion, die von den Threads ausgeführt wird
def do_work():
    for i in range(5):
        print(f"Thread {threading.get_ident()} ist am Arbeiten.")
        time.sleep(1)  # Simuliere Arbeit

# Hauptfunktion
def main():
    # Erstelle und starte zwei Threads
    thread1 = threading.Thread(target=do_work)
    thread2 = threading.Thread(target=do_work)

    thread1.start()
    thread2.start()

    # Warte, bis beide Threads beendet sind
    thread1.join()
    thread2.join()

    # Ausgabe nach Abschluss beider Threads
    print("Beide Threads sind beendet.")

# Ausführen der main-Funktion
if __name__ == "__main__":
    main()

File: C#/Multithreading.cs
using System;
using System.Threading;

public class Multithreading
{
    public Multithreading()
    {
        //Erstelle und starte zwei Threads
        Thread thread1 = new Thread(new ThreadStart(DoWork));
        Thread thread2 = new Thread(new ThreadStart(DoWork));

        thread1.Start();
        thread2.Start();

        //Warte, bis beide Threads beendet sind
        thread1.Join();
        thread2.Join();

        //Ausgabe nach Abschluss beider Threads
        Console.WriteLine("Beide Threads sind beendet.");
    }

    private void DoWork()
    {
        for (int i = 0; i < 5; i++)
        {
            Console.WriteLine($"Thread {Thread.CurrentThread.ManagedThreadId} ist am Arbeiten.");
            Thread.Sleep(1000); //Simuliere Arbeit
        }
    }

    public static void Main(string[] args)
    {
        new Multithreading();
    }
}
